File: backend/java/localpos-user-service/src/main/java/com/pds/localpos/userservice/service/impl/CustomUserDetailsService.java
package com.pds.localpos.userservice.service.impl;

import com.pds.localpos.security.service.UserDetailsServiceProvider;
import com.pds.localpos.userservice.model.LocalPosUserDetails;
import com.pds.localpos.userservice.model.User;
import com.pds.localpos.userservice.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class CustomUserDetailsService implements UserDetailsServiceProvider, UserDetailsService {

    private final UserRepository userRepository;

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        User user = userRepository.findByUsername(username)
                .orElseThrow(() -> new UsernameNotFoundException("User not found: " + username));
        return new LocalPosUserDetails(user);
    }
}

File: backend/kotlin/localpos-user-service/src/main/kotlin/com/pds/localpos/userservice/service/impl/CustomUserDetailsService.kt
package com.pds.localpos.userservice.service.impl

import com.pds.localpos.userservice.model.LocalPosUserDetails
import com.pds.localpos.userservice.repository.UserRepository
import com.pds.localpos.userservice.service.UserDetailsServiceProvider
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
class CustomUserDetailsService(
    private val userRepository: UserRepository
) : UserDetailsServiceProvider, UserDetailsService {

    override fun loadUserByUsername(username: String): UserDetails {
        val user = userRepository.findByUsername(username)
            .orElseThrow { UsernameNotFoundException("User not found: $username") }
        return LocalPosUserDetails(user)
    }
}
